src/ReservoirTimelock.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import { CompTimelock } from "@openzeppelin/mocks/compound/CompTimelock.sol";
+
+import { GenericFactory } from "src/GenericFactory.sol";
+import { ReservoirPair } from "src/ReservoirPair.sol";
+import { StablePair } from "src/curve/stable/StablePair.sol";
+
+contract ReservoirTimelock is CompTimelock(msg.sender, 7 days) {
+    function setCustomSwapFee(GenericFactory aFactory, address aPair, uint256 aSwapFee) external {
+        require(msg.sender == admin, "RT: ADMIN");
+
+        bytes memory lCalldata = abi.encodeCall(ReservoirPair.setCustomSwapFee, (aSwapFee));
+        aFactory.rawCall(aPair, lCalldata, 0);
+    }
```

Comment:
I was thinking this might be risky but i checked and we cap the fee at 2%, which feels fine. So I think we're all good